system/jlib/jsocket.cpp

```diff
@@ -4801,6 +4806,13 @@ class CSocketEpollThread: public CSocketBaseThread
             if ( !si->del && (si->sock==sock) )
                 si->del = true;
         }
+        if ( !sock || !nfy )
+        {
+            WARNLOG("EPOLL: adding fd but sock or nfy is NULL");
+            selectvarschange = true;
+            triggerselect();
```

Comment:
It was for safety / compatibility in that we do not skip the loop checking for si->sock==sock to mark those as del.
I set selectvarschanged and call triggerselect() in case any socks were marked for del.  I could add a check to see if any were changed and if not then don't set selectvarschanged and don't call triggerselect() but I didn't think it needed that fine a check as this new case would probably (hopefully) not be common.
I return true for same reason - didn't want to have any parent calling add() that checked its return throw and cause a problem - but perhaps that or even better a dbgassertex (to show traceback) is a good idea.